### ISSUE TITLE
Importing in Typescript errors ts(2614)

### ISSUE THREAD

**Issue by KieranO547** [ben-rogerson/twin.macro#77]
I finally started trying out twin (you recommended it on reddit a few weeks ago!) having issues starting up in typescript however.

I followed the installation guide for CRA, I've made two code sandboxes setting up:
Plain JS (Working): https://codesandbox.io/s/lucid-kapitsa-cjy3y?file=/src/App.js
TS (Errors): https://codesandbox.io/s/busy-moser-yxl7g?file=/src/App.tsx

As you can see the JS works fine, and the TS does display the elements styled but there is a TS error for the imports `Module '"../node_modules/twin.macro/types"' has no exported member 'tw'. Did you mean to use 'import tw from "../node_modules/twin.macro/types"' instead?ts(2614)` I'm assuming it can't find the type definitions? I know you say you've got them built in, not sure if I missed something in installation.

**Comment by rbutera** [comment]
`tw` is the default export, not an exported member IIRC

**Comment by rbutera** [comment]
We should update the typescript types definition to include {styled} and {css}, thanks for flagging these

**Comment by KieranO547** [comment]
Going off the docs shows a non-default tw export?
```import { tw, css } from "twin.macro";

const Input = ({ hasDarkHover }) => (
  <input
    css={[
      ({ hasHover }) => hasHover && tw`hover:border-black`,
      tw`border`,
      css`
        color: white;
      `,
    ]}
  />
);
export default () => <Input hasDarkHover />;```

**Comment by rbutera** [comment]
We should update the typescript types definition to include {styled} and {css}, thanks for flagging these

**Comment by AlexanderArvidsson** [comment]
@KieranO547 
To fix this, css comes from @emotion/core or your css-in-js library of choice.
Had this issue yesterday, all I had to do was this:
```
import { css } from "@emotion/core";
import tw from "twin.macro"
```
Although, it would be nice if the typings matched the documentation!

**Comment by KieranO547** [comment]
@AlexanderArvidsson Thanks! Didn't realise the default export was really all I needed lol, appreciate the help :) and thanks @rbutera + I saw the PR you're working on, much appreciated!

**Comment by KieranO547** [comment]
@ben-rogerson Can I confirm if this issue definitely closed? I've just upgraded to v1.3.0 and still see the same error in typescript when importing styled from twin.macro: 
`import tw, { styled } from 'twin.macro';` gives "has no exported member named styled".

updated sandbox on v1.3.0: https://codesandbox.io/s/busy-moser-yxl7g?file=/src/App.tsx

Since I created this issue I've been following AlexanderArvidsson suggestion of importing it directly from styled-components, but I've just found using the tw='' prop / ${tw``} function within styled-components is meaning some styles are not working properly on ie11 (unfortunately I have to support it for this project...RIP), specifically those with variables (a specific example being opacity variable in rgb colours)

Not sure if this is a separate issue in my setup or not though (FWIW if I put the tailwind classes into `className=''` the styles apply because it also sets the hex value?) 
What I get when using the tw prop:
![image](https://user-images.githubusercontent.com/17687552/83726515-87cf1c00-a63b-11ea-99bf-64fb196d38d0.png)
versus using the className: 
![image](https://user-images.githubusercontent.com/17687552/83726954-383d2000-a63c-11ea-97fa-557e6228c2e6.png)

**Comment by rbutera** [comment]
@KieranO547 

> Can I confirm if this issue definitely closed? I've just upgraded to v1.3.0 and still see the same error in typescript when importing styled from twin.macro:
> `import tw, { styled } from 'twin.macro';` gives "has no exported member named styled".

It appears that you've not followed the guide for annotating the TypeScript types for your chosen CSS-in_JS lirbary:

- [TypeScript + Styled-Components](https://github.com/ben-rogerson/twin.macro/blob/master/docs/styled-components/typescript.md)
- [TypeScript + Emotion](https://github.com/ben-rogerson/twin.macro/blob/master/docs/emotion/typescript.md)

Let us know if this resolves your issue.

<hr />

@ben-rogerson I was worried these instructions would not be visible enough if not linked to from the main readme (which is why I had initially linked them from there!), and this comment here has confirmed these suspicions. I suggest that we link them again from the readme. Thoughts?

**Comment by rbutera** [comment]
@KieranO547 regarding your hex value issue please create a separate issue as this one is now closed, or feel free to join us on the discord and discuss it there in the `#help` channel, thanks!

**Comment by KieranO547** [comment]
@rbutera FWIW I didn't realise there was new documentation since I created the issue (I used the same code sandbox and assumed nothing changed) so thats why I hadn't followed the new instruction to add the declaration, I thought it was something you guys had added to the package. This is working for me now, thank you for your help! I'll setup another issue for the hex thing :)

**Comment by luknl** [comment]
Hey, I added the [twin.d.ts](https://github.com/ben-rogerson/twin.macro/blob/master/docs/emotion/typescript.md) file. It solved the error TS errors on css and styled for exemple. But now I have errors on my props (not all for some reason):
![image](https://user-images.githubusercontent.com/9930144/101795441-d1735480-3aff-11eb-92ee-ebbaaa6cbacf.png)

Any idea why?

**Comment by tarikhamilton** [comment]
I still had issues getting this to work and some of the links here in this issue no longer worked. Thankfully `import styled from 'styled-components'` worked out fine. Docs said `styled` is re-exported from the twin.macro, so I'm most likely fine. I'm using CRA in a monorepo with styled-components and I updated the package.json. 

Not expecting a response from the maintainers, just hope this might help anyone else who wrestled with `styled` not being exported from `twin.macro`

**Comment by flavio-ever** [comment]
Hi, I'm facing this issue right now. I thought I was crazy! There is nothing talking about the problem in the doc:

https://github.com/ben-rogerson/twin.macro/blob/master/docs/styled-component-guide.md

As for the links, they are down... I opened this thread about:
https://github.com/ben-rogerson/twin.macro/discussions/571

**Comment by damaon** [comment]
Getting same issue with next-js example

**Comment by antoniobrandao** [comment]
> @KieranO547
> 
> > Can I confirm if this issue definitely closed? I've just upgraded to v1.3.0 and still see the same error in typescript when importing styled from twin.macro:
> > `import tw, { styled } from 'twin.macro';` gives "has no exported member named styled".
> 
> It appears that you've not followed the guide for annotating the TypeScript types for your chosen CSS-in-JS library:
> 
> * [TypeScript + Styled-Components](https://github.com/ben-rogerson/twin.macro/blob/master/docs/styled-components/typescript.md)
> * [TypeScript + Emotion](https://github.com/ben-rogerson/twin.macro/blob/master/docs/emotion/typescript.md)
> 
> Let us know if this resolves your issue.
> 
> @ben-rogerson I was worried these instructions would not be visible enough if not linked to from the main readme (which is why I had initially linked them from there!), and this comment here has confirmed these suspicions. I suggest that we link them again from the readme. Thoughts?

These links are dead, any chance of providing new ones?

**Comment by AlexanderArvidsson** [comment]
@antoniobrandao The authors have switched to mostly just having examples for each library or framework you are using.
It's impossible for them to provide generic documentation that works for all libraries, so examples are much better.
For example, here is the link for Next.JS emotion example: https://github.com/ben-rogerson/twin.examples/tree/master/next-emotion-typescript

Keep in mind that every time emotion or any other library updates their typings, twin.macro has to do the same, and so do you.
From experience, most of the problems in this issue is because people have not kept their twin.d.ts updated.
Please make sure you have the latest version of the twin.d.s file.

Just go to the example of the framework / build tool you are using in the readme, and copy the twin.d.ts file in that example.
If there's further issues, you should raise a new issue for that specific example with a reproducible demo.

(Disclaimer: I am not a contributor to this package, this is just my thoughts on this)
